pkg/api/mlflow/service/artifact/validators.go

```diff
@@ -10,5 +12,9 @@ func ValidateListArtifactsRequest(req *request.ListArtifactsRequest) error {
 	if req.RunID == "" && req.RunUUID == "" {
 		return api.NewInvalidParameterValueError("Missing value for required parameter 'run_id'")
 	}
+
+	if strings.Contains(req.Path, ".") || strings.Contains(req.Path, "..") {
```

Comment:
what about a path like `my.directory.containing.dots/whatever`?
I'd suggest using a regex and checking specifically for directories that have _only_ 1 or 2 dots

Reply:
I mean url.JoinPath when you append the experiment IDs and run IDs later — but also why are you trimming the Host here? and why not just trim the slashes from the path and use u.String()?
- hm, probably this is something old what you can see. let me a bit simplify this case.

Reply:
@jgiannuzzi about `url.JoinPath` . the problem and the point why I don't like to use it, because it returns `err` for such a simple operation, so it is really annoying to check it instead of to construct everything in just one line using `sprintf`.